lib/docker/api.ex

```diff
@@ -22,16 +22,48 @@ defmodule Testcontainers.Docker.Api do
     end
   end
 
-  def pull_image(image, conn) when is_binary(image) do
-    case Api.Image.image_create(conn, fromImage: image) do
-      {:ok, %Tesla.Env{status: 200}} ->
-        {:ok, nil}
+  def list_images(conn) do
+    case DockerEngineAPI.Api.Image.image_list(conn) do
+      {:ok, %DockerEngineAPI.Model.ErrorResponse{} = error} ->
+        {:error, {:failed_to_list_images, error}}
+
+      {:ok, list} ->
+        list
+        # |> Enum.map(fn x -> Map.get(x, :RepoTags) end)
+        |> Enum.flat_map(fn x -> Map.get(x, :RepoTags, []) || [] end)
+        |> Enum.uniq()
+        |> Enum.sort()
+    end
+  end
 
-      {:error, %Tesla.Env{status: other}} ->
-        {:error, {:http_error, other}}
+  def ping(conn) do
+    {:ok, %{body: body}} = DockerEngineAPI.Api.System.system_ping(conn)
+    body
+  end
 
-      {:ok, %DockerEngineAPI.Model.ErrorResponse{} = error} ->
-        {:error, {:failed_to_pull_image, error}}
+  def version(conn) do
+    {:ok, info} = DockerEngineAPI.Api.System.system_version(conn)
+    info
+  end
+
+  def pull_image(config, conn) when is_binary(config.image) do
```

Comment:
is config a map or a struct? What's the API of it?

Reply:
if you pass something that doesnt have a image struct, you'll get a `FunctionClauseError`. Its better to handle this explicitly in the function body